ttnn/cpp/ttnn/tensor/types.hpp

```diff
@@ -424,35 +424,38 @@ struct MultiDeviceHostStorage {
         std::vector<OwnedBuffer> buffers;
         std::vector<Shape> shapes;
         mutable std::mutex mtx;
+
+        friend void swap(MultiDeviceHostStorage& first, MultiDeviceHostStorage& second) {
+            std::scoped_lock lock(first.mtx, second.mtx);
+            // enable ADL (not necessary, but good practice)
```

Comment:
This comment isn't accurate.`using std::swap;` doesn't "enable ADL", it defines a fallback in case ADL fails to locate a viable candidate.

Reply:
Btw probably I don't need it here at all. 
Need of the lock removes all fun here :(

Reply:
If it weren't for the lock, the copy and move constructors, and copy and move assignment could all be `= default;` as well. Yes, locks take away all the fun 🙃